File: 02_Patterns/Pattern_11/Pattern_11.py
class Solution:
    def printTriangle(self, N):
        # Outer loop for number of rows/lines
        for i in range(1, N+1):

            if i % 2 == 0:
                start = 0   # If i is even, start with 0
            else:
                start = 1   # If i is odd, start with 1

            # Inner loop for columns to print the values in the row
            for j in range(1, i+1):

                print(start, end=" ")

                # toggle start from 1 to 0 and 0 to 1
                start = 1-start

            print()


# {
# Driver Code Starts
# Initial Template for Python 3
if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())
        ob = Solution()
        ob.printTriangle(N)
# } Driver Code Ends

File: 02_Patterns/Pattern_11/Pattern_11.js
function zeroOnesTrianglePattern(N) {
  // Outer loop for number of rows/lines
  for (let i = 1; i <= lines; i++) {
    let line = '';

    // Inner loop for columns to print the values in the row
    for (let j = 1; j <= i; j++) {
      // If the sum of line index and column index is even, print 1, else print 0
      line += (i + j) % 2 === 0 ? '1 ' : '0 ';
    }

    console.log(line);
  }
}

// Example usage:
zeroOnesTrianglePattern(5);
